### PR TITLE
Move compilation into /lib

### PR DESCRIPTION
**Pull request by roc**
### Adds in test fixtures
- English and Arabic `src` folders (culled from EVW)
- any outputted files are ignored in `.gitignore`
### lib/compile.js
- Adds callback to the end of compilation
  - so we can play around with post-walk actions
  - also good for tests passing `done()`
- Removes dependency on `underscore-deep-extend`
  - Object.assign is fine here
- Separates `directory` function out to reduce nesting in callbacks
- Adds tests
  - dumb tests, there to bootstrap in an end to end fixture related test
  - can get more granular as the compiler code becomes more granular
### Removes use of `rimraf`
- we don't need it, as `mkdirp` is overwriting our folders
- `dirs.delete` left in, ready to be deprecated/removed unless there's some terrible need for it that I've missed
### Remove `underscore` dependency
- iterating and combining objects using native `Object` methods

### CODE REVIEW

**Comment by joefitter** [comment]
nice, LGTM - @JoeChapman ?

**Comment by roc** [comment]
@joefitter @JoeChapman I'm doing a slightly ridiculous thing in the tests (but it's only in the tests) here: https://github.com/UKHomeOffice/hof-transpiler/pull/24/files#diff-bf854085839aa797e5cd956dbc1cc6d7R35

Feel like this is still a good patch before someone comes along and does a clarifying refactor at some point

**Comment by joefitter** [comment]
I did see that - so you are checking the contents of the compiled file? Seems fine for tests as we want to isolate functionality as much as poss.

**Comment by roc** [comment]
@joefitter Yeah checking the folders exist, then checking the compiled `default.json` files look correct. Nothing fancy. The compile callback is fired immediately after the `walker.walk` sequence has finished, but the read/stat-ing seems to take a little longer, hence the slightly resigned timeout wrapper.

File: lib/compile.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+            fs.writeFileSync(writeFileStream[key].path, strung, 'utf8');
+            next();
+          });
+        });
+
+        writeFileStream[key].end('}');
+      });
+    });
+  });
+  if (cb) {
```
Comment (joefitter) [inline lib/compile.js]: @roc could it be that we are calling cb outside of the walker.on end event?

Comment (roc) [reply lib/compile.js]: @joefitter I was just writing a very long explanation of why that's not the case, but I think I must've fudged the `cb` placement originally, and actually moving it to the bottom of the `'end'` statement works a treat. Who would've thought?!

Comment (joefitter) [reply lib/compile.js]: always something simple ;)

Comment (roc) [reply lib/compile.js]: here we go https://github.com/UKHomeOffice/hof-transpiler/pull/24/commits/f624d73c48e0c6f0867b3b721883279ed5b79395

**Comment by joefitter** [comment]
boom - lgtm @JoeChapman ?

**Comment by JoeChapman** [comment]
looking

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,5 @@
 .idea/*
+test/fixtures/ar
+test/fixtures/en
```
Comment (JoeChapman) [inline .gitignore:3]: maybe just `test/fixtures/**`?

**Comment by JoeChapman** [comment]
LGTM

File: .gitignore (hunk 2 above)
Comment (joefitter) [reply .gitignore:3]: test/fixtures/src is needed - merging

**Comment by roc** [comment]
🎉 psyche🎉

**Pull request merged**